File: contest/src/Domain/Department.java
package Domain;

/**
 * Created by Sebi on 10/7/2016.
 */

import java.io.Serializable;

/**
 * Gestioneaza obiectele de tip Department
 */
public class Department implements HasId<Integer>, Serializable {
    private Integer id;
    private String name;
    private Integer numberOfPlaces;

    /**
     * Constructor
     * @param id - Integer
     * @param name - String
     * @param numberOfPlaces - Integer
     */
    public Department(Integer id, String name, Integer numberOfPlaces){
        this.id = id;
        this.name = name;
        this.numberOfPlaces = numberOfPlaces;
    }

    /**
     * Getters
     */
    public Integer getId(){
        return id;
    }

    public String getName(){
        return name;
    }

    public Integer getNumberOfPlaces(){
        return numberOfPlaces;
    }

    /**
     * Setters
     */
    public void setId(Integer id){
        this.id = id;
    }

    public void setName(String name){
        this.name = name;
    }

    public void setNumberOfPlaces(Integer numberOfPlaces){
        this.numberOfPlaces = numberOfPlaces;
    }

    /*
    To string method
     */
    @Override
    public String toString(){
        return id + " " + name + " " + numberOfPlaces;
    }
}

File: contestC#/contest/contest/Domain/Department.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace contest.Domain
{
    public class Department : HasID<int>
    {
        //attributes
        private int id;
        private string name;
        private int numberOfPlaces;

        //getters and setters
        public int Id
        {
            get
            {
                return id;
            }

            set
            {
                id = value;
            }
        }

        public string Name
        {
            get
            {
                return name;
            }

            set
            {
                name = value;
            }
        }

        public int NumberOfPlaces
        {
            get
            {
                return numberOfPlaces;
            }

            set
            {
                numberOfPlaces = value;
            }
        }

        public string IdName => $"{Id} {Name}"; //read only property

        //Constructor
        public Department(int id, string name, int numberOfPlaces)
        {
            this.id = id;
            this.name = name;
            this.numberOfPlaces = numberOfPlaces;
        }

        //to string method
        public override string ToString()
        {
            return id + " " + name + " " + numberOfPlaces;
        }

        //equals method
        public override bool Equals(object obj)
        {
            if (obj == null)
            {
                return false;
            }
            if (!(obj is Department))
            {
                return false;
            }
            Department d = (Department)obj;
            return id == d.id;
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }
    }
}
